src/index.js

```diff
@@ -473,6 +496,256 @@ const initialize = async () => {
+     * Personal Sign
+     */
+
+  personalSignTest.onclick = async () => {
+    const exampleMessage = 'Sample message for personal sign test'
+    try {
+      const from = accounts[0]
+      const msg = `0x${Buffer.from(exampleMessage, 'utf8').toString('hex')}`
+      const result = await ethereum.request({
+        method: 'personal_sign',
+        params: [msg, from, 'Example password'],
+      })
+      console.log('recovering!!!')
+      const recoveredAddr = recoverPersonalSignature({
+        'data': msg,
+        'sig': result,
+      })
+      if (recoveredAddr === from) {
+        console.log(`SigUtil Successfully verified signer as ${recoveredAddr}`)
+      } else {
+        console.log(`SigUtil Failed to verify signer when comparing ${recoveredAddr} to ${from}`)
+        console.log(`Failed comparing ${recoveredAddr} to ${from}`)
+      }
+    } catch (err) {
+      console.error(err)
+    }
+  }
+
+  /**
+   * Personal ecRecover Test
+   */
+  personalRecoverTest.onclick = async () => {
+    const exampleMessage = 'Sample message for personal ec Recover Test'
+    try {
+      const from = accounts[0]
+      const msg = `0x${Buffer.from(exampleMessage, 'utf8').toString('hex')}`
+      const result = await ethereum.request({
+        method: 'personal_sign',
+        params: [msg, from, 'Example password'],
+      })
+      console.log(result)
+      console.log('recovering!!!')
+      const ecRecoverAddr = await ethereum.request({
+        method: 'personal_ecRecover',
+        params: [msg, result],
+      })
+      if (ecRecoverAddr === from) {
+        console.log(`Successfully ecRecovered signer as ${ecRecoverAddr}`)
+      } else {
+        console.log(`Failed to verify signer when comparing ${ecRecoverAddr} to ${from}`)
+      }
+    } catch (err) {
+      console.error(err)
+    }
+  }
+
+  /**
+   * Sign Typed Data Test
+  */
+  signTypedDataTest.onclick = async () => {
+    const msgParams = [
+      {
+        type: 'string',
+        name: 'Message',
+        value: 'Hi, Alice!',
+      },
+      {
+        type: 'uint32',
+        name: 'A number',
+        value: '1337',
+      },
+    ]
+
+    try {
+      const from = accounts[0]
+      const signTypedResult = await ethereum.request({
+        method: 'eth_signTypedData',
+        params: [msgParams, from],
+      })
+      console.log(signTypedResult)
+      console.log('recovering!!!')
+      const recoveredAddr = recoverTypedSignatureLegacy({
+        'data': msgParams,
+        'sig': signTypedResult,
+      })
+      console.log(recoveredAddr)
+      if (toChecksumAddress(recoveredAddr) === toChecksumAddress(from)) {
+        console.log(`Successfully verified signer as ${recoveredAddr}`)
+      } else {
+        console.log(`Failed to verify signer when comparing ${recoveredAddr} to ${from}`)
+      }
+    } catch (err) {
+      console.log(err)
+    }
+  }
+
+  /**
+   * Sign Typed Data Version 3 Test
+   */
+  signTypedDataV3Test.onclick = async () => {
```

Comment:
`signTypedDataV3Test` is redundant with the existing `signTypedData` button in this file. We should remove `signTypedDataV3Test` completely. And rename the existing `signTypedData` button to `signTypedDataV3`.